cpp/src/arrow/engine/substrait/function_test.cc

```diff
@@ -487,7 +497,13 @@ TEST(FunctionMapping, AggregateCases) {
        float64(),
        "[2]",
        "[1.5, 3]",
-       float64()}};
+       float64()},
+      {{kSubstraitAggregateGenericFunctionsUri, "count"},
+       {"[1, 2, 30]"},
+       {int8()},
+       "[3]",
+       "[2, 1]",
+       int64()}};
```

Comment:
Unfortunately, the count function takes in a [single optional enum argument](https://github.com/substrait-io/substrait/blob/main/extensions/functions_aggregate_generic.yaml#L8).  I have no idea what the purpose of this argument is.  However, in Substrait, "optional arguments" should still be included in the protobuf.

I realize now that I did not account for this in the other aggregate functions too :(.

Reply:
I'm new to substrait. Maybe it means the enumeration arguments.
https://github.com/substrait-io/substrait/blob/main/text/simple_extensions_schema.yaml#L94
And here may have some clue https://substrait.io/expressions/scalar_functions/#optional-enumeration-properties